src/Customer.hs

```diff
@@ -0,0 +1,64 @@
+-- SPDX-FileCopyrightText: 2022 Serokell <https://serokell.io/>
+--
+-- SPDX-License-Identifier: Apache-2.0
+
+{-# LANGUAGE RecordWildCards #-}
+
+module Customer
+  ( module Customer
+  , ReportPushMetricsBody(..)
+  , TrackAnonymusEventBody(..)
+  , TrackCustomerEventBody(..)
+  ) where
+
+import Customer.Events.API (api)
+import Customer.Events.Types.ReportPushMetrics (ReportPushMetricsBody(..), defaultReportPushMetrics)
+import Customer.Events.Types.TrackAnonymusEvent (TrackAnonymusEventBody(..), defaultTrackAnonymusEvent)
+import Customer.Events.Types.TrackCustomerEvent (TrackCustomerEventBody(..), defaultTrackCustomerEvent)
+import Data.Text (Text)
+import qualified Network.HTTP.Client as HTTP
+import Servant.API
+import Servant.Client
+
+host :: BaseUrl
+host = BaseUrl Https "track.customer.io" 443 ""
+
+data Env = MkEnv
+  { authtoken :: BasicAuthData
+  , clientEnv :: ClientEnv
+  }
+
+mkEnv :: BasicAuthData -> HTTP.Manager -> Env
+mkEnv authtoken httpManager = MkEnv {..}
+  where
+    clientEnv = mkClientEnv httpManager host
+
+trackCustomerEventC :: BasicAuthData -> Text -> TrackCustomerEventBody -> ClientM ()
+trackAnonymusEventC :: BasicAuthData -> TrackAnonymusEventBody -> ClientM ()
+reportPushMetricsC  :: ReportPushMetricsBody -> ClientM ()
+
+trackCustomerEventC
+  :<|> trackAnonymusEventC
+  :<|> reportPushMetricsC
+  = client api
+
+trackCustomerEventC' :: Env -> Text -> TrackCustomerEventBody -> IO (Either ClientError ())
+trackCustomerEventC' MkEnv{..} identifier body = do
+  runClientM (trackCustomerEventC authtoken identifier body) clientEnv
+
+trackAnonymusEventC' :: Env -> TrackAnonymusEventBody -> IO (Either ClientError ())
+trackAnonymusEventC' MkEnv{..} body = do
+  runClientM (trackAnonymusEventC authtoken body) clientEnv
+
+reportPushMetricsC' :: Env -> ReportPushMetricsBody -> IO (Either ClientError ())
+reportPushMetricsC' MkEnv{..} body = do
+  runClientM (reportPushMetricsC body) clientEnv
+
+trackCustomerEvent :: Env -> Text -> Text -> IO (Either ClientError ())
```

Comment:
Can you justify why helpers with `default*Event` bodies can be useful? I can think of a use-case for them when using this library as an SDK

Reply:
I'm not sure, that they are useful, but `default*Event` are useful because user can override part of fields and don't touch other. I can remove "the most default" helpers, but in that case I need to know that they aren't useful. Basically, I just applied structure from `mixpanel-client`, that Domen mentioned at the start.

Reply:
The structure is good, but I can't think of a use-case for `default*Event` helpers